src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -32,39 +34,59 @@ func (client *Client) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 // connetion and starts piping the data over a new net.Conn obtained from the
 // given dial function.
 func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
-	if req.Method != "CONNECT" {
+	var err error
+
+	// intercept can only by used for CONNECT requests.
+	if req.Method != httpConnectMethod {
 		panic("Intercept used for non-CONNECT request!")
 	}
 
-	// Hijack underlying connection
-	clientConn, _, err := resp.(http.Hijacker).Hijack()
-	if err != nil {
+	// Hijacking underlying connection.
+	var clientConn net.Conn
+	if clientConn, _, err = resp.(http.Hijacker).Hijack(); err != nil {
```

Comment:
I think so, the useful thing I see here is the comparison that follows the declaration rather than the declaration itself:

```
if foo, err = baz(); err != nil {
  // Do something with err
}
```

Instead of:

```
foo, err = baz()

if err != nil {
  // Do something with err
}
```

The first case does not (easily) allow anything to happen between the error assignation and the comparison it can be easily read like a single *block*, while in the latter there's always a chance of putting more instructions in between:

```
foo, err = baz()

bar, err = baroo()

if err != nil {
  // You didn't catch the first error!
}
```